cmd/agent/app/agent.go

```diff
@@ -123,7 +123,7 @@ func setupControllers(mgr controllerruntime.Manager, opts *options.Options, stop
 		ClusterLeaseRenewIntervalFraction: opts.ClusterLeaseRenewIntervalFraction,
 	}
 	if err := clusterStatusController.SetupWithManager(mgr); err != nil {
-		klog.Fatalf("Failed to setup cluster status controller: %v", err)
```

Comment:
For example, if SetupWithManager failed, what we really care about is the reason why the process start failed. And the error will be printed in the logs. 
Klog.Fatal will print a stack trace of all running goroutines. And we don't care about this info(which maybe contain hundreds of lines of info, and this too much info will prevent us from directly finding the error reason).

Reply:
> which maybe contain hundreds of lines of info, 

An example?

Reply:
I change this log print from `Infof` to `Fatal`
https://github.com/karmada-io/karmada/blob/00307f25c442f96ed54ea9e233e1cb288931feb4/cmd/agent/app/agent.go#L70
and we'll see that if we use `klog.Fatalf` to print fatal logs, there will be hundreds of lines of goroutine info, that's not necessary, we just need to know the reason why the progress exist. we don't care about the Goroutine info, which contains too much info that is not helpful for us debugging the progress exist  not normally.
```
 ./karmada-agent --cluster-name temp
F1222 10:07:11.876212 1626514 agent.go:70] karmada-agent version: version.Info{GitVersion:"v0.5.0-1010-g00307f25-dirty", GitCommit:"00307f25c442f96ed54ea9e233e1cb288931feb4", GitTreeState:"dirty", BuildDate:"2021-12-22T02:06:30Z", GoVersion:"go1.17.2", Compiler:"gc", Platform:"linux/amd64"}
goroutine 1 [running]:
k8s.io/klog/v2.stacks(0x1)
	/root/go/src/github.com/karmada-io/karmada/vendor/k8s.io/klog/v2/klog.go:1026 +0x8a
k8s.io/klog/v2.(*loggingT).output(0x2ffdec0, 0x3, {0x0, 0x0}, 0xc0001e6ee0, 0x0, {0x24b58e0, 0xc000088bd0}, 0x0, 0x0)
	/root/go/src/github.com/karmada-io/karmada/vendor/k8s.io/klog/v2/klog.go:975 +0x63d
k8s.io/klog/v2.(*loggingT).printf(0x28, 0x1f8893c, {0x0, 0x0}, {0x0, 0x0}, {0x1d427ab, 0x19}, {0xc000088bd0, 0x1, ...})
	/root/go/src/github.com/karmada-io/karmada/vendor/k8s.io/klog/v2/klog.go:753 +0x1e5
k8s.io/klog/v2.Fatalf(...)
	/root/go/src/github.com/karmada-io/karmada/vendor/k8s.io/klog/v2/klog.go:1514
github.com/karmada-io/karmada/cmd/agent/app.run({0x1fcf390, 0xc000339d40}, {0x1fb3938, 0xc000562b20}, 0xc0005b7e00)
	/root/go/src/github.com/karmada-io/karmada/cmd/agent/app/agent.go:70 +0x128
github.com/karmada-io/karmada/cmd/agent/app.NewAgentCommand.func1(0xc000032a00, {0xc00030e420, 0x2, 0x2})
	/root/go/src/github.com/karmada-io/karmada/cmd/agent/app/agent.go:48 +0x6d
github.com/spf13/cobra.(*Command).execute(0xc000032a00, {0xc000122070, 0x2, 0x2})
	/root/go/src/github.com/karmada-io/karmada/vendor/github.com/spf13/cobra/command.go:856 +0x60e
github.com/spf13/cobra.(*Command).ExecuteC(0xc000032a00)
	/root/go/src/github.com/karmada-io/karmada/vendor/github.com/spf13/cobra/command.go:974 +0x3bc
github.com/spf13/cobra.(*Command).Execute(...)
	/root/go/src/github.com/karmada-io/karmada/vendor/github.com/spf13/cobra/command.go:902
main.main()
	/root/go/src/github.com/karmada-io/karmada/cmd/agent/main.go:29 +0x4f

goroutine 19 [chan receive]:
k8s.io/klog/v2.(*loggingT).flushDaemon(0x0)
	/root/go/src/github.com/karmada-io/karmada/vendor/k8s.io/klog/v2/klog.go:1169 +0x6a
created by k8s.io/klog/v2.init.0
	/root/go/src/github.com/karmada-io/karmada/vendor/k8s.io/klog/v2/klog.go:420 +0xfb

goroutine 20 [sleep]:
time.Sleep(0x6fc23ac00)
	/usr/local/go/src/runtime/time.go:193 +0x12e
sigs.k8s.io/controller-runtime/pkg/log.init.0.func1()
	/root/go/src/github.com/karmada-io/karmada/vendor/sigs.k8s.io/controller-runtime/pkg/log/log.go:63 +0x38
created by sigs.k8s.io/controller-runtime/pkg/log.init.0
	/root/go/src/github.com/karmada-io/karmada/vendor/sigs.k8s.io/controller-runtime/pkg/log/log.go:62 +0x25

goroutine 100 [select]:
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0x0, {0x1f95c00, 0xc00004c180}, 0x1, 0xc0001141e0)
	/root/go/src/github.com/karmada-io/karmada/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:167 +0x13b
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0x0, 0x12a05f200, 0x0, 0x0, 0x0)
	/root/go/src/github.com/karmada-io/karmada/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x89
k8s.io/apimachinery/pkg/util/wait.Until(...)
	/root/go/src/github.com/karmada-io/karmada/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:90
k8s.io/apimachinery/pkg/util/wait.Forever(0x0, 0x0)
	/root/go/src/github.com/karmada-io/karmada/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:81 +0x28
created by k8s.io/component-base/logs.InitLogs
	/root/go/src/github.com/karmada-io/karmada/vendor/k8s.io/component-base/logs/logs.go:58 +0x79

goroutine 102 [syscall]:
os/signal.signal_recv()
	/usr/local/go/src/runtime/sigqueue.go:169 +0x98
os/signal.loop()
	/usr/local/go/src/os/signal/signal_unix.go:24 +0x19
created by os/signal.Notify.func1.1
	/usr/local/go/src/os/signal/signal.go:151 +0x2c

goroutine 103 [chan receive]:
k8s.io/apiserver/pkg/server.SetupSignalContext.func1()
	/root/go/src/github.com/karmada-io/karmada/vendor/k8s.io/apiserver/pkg/server/signal.go:48 +0x2b
created by k8s.io/apiserver/pkg/server.SetupSignalContext
	/root/go/src/github.com/karmada-io/karmada/vendor/k8s.io/apiserver/pkg/server/signal.go:47 +0xe7
```
take a real example: 
https://github.com/karmada-io/karmada/blob/00307f25c442f96ed54ea9e233e1cb288931feb4/cmd/agent/app/agent.go#L129
What we care about is the error that why SetupWithManager failed, and the error will be printed in the log, we don't care about Goroutine info, that's too much.